backend/src/main/java/de/bund/digitalservice/ris/caselaw/domain/TextCheckService.java

```diff
@@ -59,19 +60,22 @@ public List<Match> checkWholeDocumentationUnit(UUID id)
     Documentable documentable = documentationUnitRepository.findByUuid(id);
 
     if (documentable instanceof DocumentationUnit documentationUnit) {
-      List<Match> allMatches = new ArrayList<>();
-
-      for (CategoryType type : CategoryType.values()) {
-        try {
-          TextCheckCategoryResponse response = checkCategory(documentationUnit, type);
-          if (response == null) {
-            continue;
-          }
-          allMatches.addAll(response.matches());
-        } catch (Exception e) {
-          log.error("Could not process category", e);
-        }
-      }
+      List<Match> allMatches = Collections.synchronizedList(new ArrayList<>());
+
+      stream(CategoryType.values())
+          .parallel()
```

Comment:
good idea.
one wishes: the log message should have more informations (uuid of the documentation unit, category type)